### PR TITLE
Add Homework # 19.1. soluion Mars Rover “Curiosity” photos.

### PR DESCRIPTION
**Pull request by barchiso**
Add Homework # 19.1. solution Mars Rover “Curiosity” photos.

Instead 'DEMO_KEY' used my registered 'API_KEY' 
with  'DEMO_KEY' receive - 'HTTP Error: 429 Client Error: Too Many Requests for URL.'

### CODE REVIEW

File: Lesson_19/homework_19_1.py [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+            _log.error(error)
+        except requests.exceptions.ConnectionError as error:
+            error = f'Connection Error: {error}'
+            _log.error(error)
+        except requests.exceptions.Timeout as error:
+            error = f'Timeout Error: {error}'
+            _log.error(error)
+        except requests.exceptions.RequestException as error:
+            error = f'Request failed: {error}'
+            _log.error(error)
```
Comment (antohaUa) [inline Lesson_19/homework_19_1.py]: You have the same errors handling in second call - better to organize single function aka do_request and use for both methods.

File: Lesson_19/homework_19_1.py [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+    def __init__(self, url, params):
+        """Initialize the MarsRoverPhotos instance.
+
+        Args:
+            url (str): The API endpoint URL.
+            params (dict): Parameters to be sent with the API request.
+        """
+        self.url = url
+        self.params = params
+        self.timeout = 5
```
Comment (antohaUa) [inline Lesson_19/homework_19_1.py]: this better to set as named init attribute

File: Lesson_19/homework_19_1.py [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+        except requests.exceptions.ConnectionError as error:
+            error = f'Connection Error: {error}'
+            _log.error(error)
+        except requests.exceptions.Timeout as error:
+            error = f'Timeout Error: {error}'
+            _log.error(error)
+        except requests.exceptions.RequestException as error:
+            error = f'Request failed: {error}'
+            _log.error(error)
+
```
Comment (antohaUa) [inline Lesson_19/homework_19_1.py:129]: In general code looks too heavy for such simple solution - try to optimize

**Review by antohaUa — CHANGES_REQUESTED** [review]

File: Lesson_19/homework_19_1.py (hunk 1 above)
Comment (barchiso) [reply Lesson_19/homework_19_1.py]: Fixed in commit 31dfb0b758979996cd1a5d4f706a0d589d4b24f9

File: Lesson_19/homework_19_1.py (hunk 2 above)
Comment (barchiso) [reply Lesson_19/homework_19_1.py]: Fixed in commit 31dfb0b758979996cd1a5d4f706a0d589d4b24f9

File: Lesson_19/homework_19_1.py (hunk 3 above)
Comment (barchiso) [reply Lesson_19/homework_19_1.py:129]: Fixed in commit 31dfb0b758979996cd1a5d4f706a0d589d4b24f9

File: Lesson_19/homework_19_1.py [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+    def save_image(self, img_url, index):
+        """Save an image locally from its URL.
+
+        Args:
+            img_url (str): URL of the image to download.
+            index (int): Index for naming the file.
+        """
+        # Send a request to download the image
+        img_response = self.do_request(img_url)
+        img_response.raise_for_status()
```
Comment (antohaUa) [inline Lesson_19/homework_19_1.py]: Looks as redundant line

**Review by antohaUa — APPROVED** [review]

File: Lesson_19/homework_19_1.py (hunk 4 above)
Comment (barchiso) [reply Lesson_19/homework_19_1.py]: remove this lone 2b469becf6d58933a23e531e12801c7cdb57f2d7

**Pull request merged**